### PR TITLE
Fixes issue #10

### PR DESCRIPTION
**Pull request by thomazz-nl**
Issue was caused by two things:
1. Google has some bad encoded characters in the response of get_embed_page(). I re-encoded to ascii while ignoring bad characters and decoded it again to clean the response.
2. The regex to find the javascript https://www.youtube.com/s/player/xxxxxxxx/player_ias.vflset/xx_XX/base.js in the embed response was not working. My new regex works.

This is my first GitHub pull request/contribution to someone elses project. Please check carefully and let me know if I need to improve anything or do anything differently.

### CODE REVIEW

File: resources/lib/youtube_plugin/youtube/helper/video_info.py [hunk 1]
```diff
@@ -605,14 +605,13 @@ def _normalize(url):
             return _normalize(javascript_url)
 
         page_result = self.get_embed_page(video_id)
-        html = page_result.get('html')
+        html = page_result.get('html').encode('ascii', 'ignore').decode('ascii')
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/youtube/helper/video_info.py]:
```
        html = page_result.get('html')
        html = html.encode('utf8', 'ignore')
        html = html.decode('utf8')
```

File: resources/lib/youtube_plugin/youtube/helper/video_info.py [hunk 2]
```diff
@@ -605,14 +605,13 @@ def _normalize(url):
...
 
         if not html:
             return ''
 
-        found = re.search(
-            r'<script src="(?P<url>[^"]+?)"\s*name="player_[^/]+?/base"\s*>\s*</script\s*>', html
-        )
+        found = re.search(r'<script src="(?P<url>[^"]+?player_[^"]+?/base\.js)"\s*>\s*</script\s*>', html)
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/youtube/helper/video_info.py]:
```
        found = re.search(
            r'<script src="(?P<url>[^"]+?)"\s*name="player_[^/]+?/base"\s*>\s*</script\s*>', html
        )
        if not found:
            found = re.search(r'<script src="(?P<url>[^"]+?player_[^"]+?/base\.js)"\s*>\s*</script\s*>', html)
```

**Review by anxdpanic — CHANGES_REQUESTED** [review]
Thank you for looking into this!
Can you try the changes I suggested, your current iteration breaks one of my installs. The requested changes should hopefully cover everybody.

**Comment by thomazz-nl** [comment]
Yes, your suggested changes fix the issue too. As content from the youtube page should be utf-8, this actually makes more sense.  The double regex check works too. I'm checking how I can merge your edits in my PR or how this works. Looking for "approve" button. Edit: sorry for the extra notifications, in the meantime I figured out how to merge your suggestions and reapply for a review :).

**Review by anxdpanic — APPROVED** [review]

**Comment by anxdpanic** [comment]
Thank you again!

If you happen to have the inclination, could you dump the embed page html to a file and attach it to a message on this PR? No worries if not.

Will then be able to see about simplifying the regex.

**Comment by thomazz-nl** [comment]
You're welcome. I generated the embed html page using this snippet:
`with io.open(r'C:\Users\User\Desktop\youtube_embed.html', 'w', encoding='utf-8', errors='ignore') as file:
    file.write(result.text)`

Here is the resulting html page: [youtube_embed_redacted.zip](https://github.com/anxdpanic/plugin.video.youtube/files/5966197/youtube_embed_redacted.zip). It's as original as possible (unformatted), but since reproducing the bug required me to be logged in and play an item from a playlist, the file contained some possibly identifying values. I carefully redacted these as they should be of no importance regarding the regex.

Maybe it's easier to use a regex to grab the base.js URL from the javascript "ytcfg" variable section? Given that Google doesn't change these often... Can you check if [this regex](https://github.com/anxdpanic/plugin.video.youtube/compare/master...thomazz-nl:master) works for all your test-cases?

**Pull request merged**